### ISSUE TITLE
Invoke-SPDSCCommand in SharePointDsc.Util.psm1 memory leak

### ISSUE THREAD

**Issue by adruet29** [PowerShell/SharePointDsc#1037]
#### Details of the scenario you tried and the problem that is occurring
I was using 2.3.0.0 version before upgrading my test environnment to 3.2.0.0
After upgrading to 3.2.0.0 I noticed an issue where my sharepoint servers would run out of memory after a couple of hours (arround 10 hours).
This happened even if the SharepointDsc module had no modification to the existing environment.
We have a very high configuration check frequency on test environment (every 15 minutes) to have the results of configuration changes very quickly.
Running the following powershell command would show many wsmprovhost.exe processes which are an indication of many powershell sessions running on the machine:
Get-Process -Name wsmprovhost

#### Suggested solution to the issue
After doing some debug, I found that the function Invoke-SPDSCCommand of the SharepointDsc.Util module was never removing the powershell session as it was the case before in my 2.3.0.0 version.
It looks like it is a wrong copy/paste of the section with local credential (which does not use a powershell) to the section where the powershell was in used which causes the issue.
Simply putting the results of the invoke-command in a variable, removing the powershell session, and then returning the result corrects the issue.
I will issue a pull request to correct this.
#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
Many configurations will have the issue as to test the compliance, many of them use an invoke-spdsccommand with credentials, thus launching a "local" remote powershell session which is never closed
Most of production environments will not see the issue as the configuration check frequency is probably higher than 15 minutes and the Invoke-SPDSCCommand as an idle timeout set to 60000. So the wsmprovhost process will mostly be killed before causing a memory issue.

# insert configuration here
```

#### The operating system the target node is running
OsName               : Microsoft Windows Server 2016 Datacenter
OsOperatingSystemSKU : DatacenterServerEdition
OsArchitecture       : 64-bit
WindowsBuildLabEx    : 14393.2485.amd64fre.rs1_release.180827-1809
OsLanguage           : en-US
OsMuiLanguages       : {en-US}

#### Version of SharePoint that is used (e.g. SharePoint 2016)
SharePoint 2016
#### Version and build of PowerShell the target node is running
Name                           Value                                                                                                    
----                           -----                                                                                                    
PSVersion                      5.1.14393.2485                                                                                           
PSEdition                      Desktop                                                                                                  
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                  
BuildVersion                   10.0.14393.2485                                                                                          
CLRVersion                     4.0.30319.42000                                                                                          
WSManStackVersion              3.0                                                                                                      
PSRemotingProtocolVersion      2.3                                                                                                      
SerializationVersion           1.1.0.1      

#### Version of the DSC module that was used ('dev' if using current dev branch)
3.2.0.0 dev branch

**Comment by sguilbault-sherweb** [comment]
I added a quick fir with pull request 1042

**Comment by ykuijs** [comment]
Closed via PR #1042
